pkg/kubelet/dockershim/docker_container.go

```diff
@@ -415,7 +415,7 @@ func (ds *dockerService) ContainerStatus(_ context.Context, req *runtimeapi.Cont
 
 	labels, annotations := extractLabels(r.Config.Labels)
 	imageName := r.Config.Image
-	if len(ir.RepoTags) > 0 {
+	if ir != nil && len(ir.RepoTags) > 0 {
```

Comment:
Instead of having adding an `ir != nil`, could we add the `if len(ir.RepoTags)... { imageName = ...` component to the block from 349-358 where we first assign `ir, err := ds.client.InspectImageByID(r.Image)` and then check if there's a failure? We could add a block, executed only if `err == nil` which executes the `if len(ir.RepoTags)... { imageName = ...` component?

In my mind, the advantage of my proposed change would be that we are only referencing `ir` in one part of the function. Checking `ir != nil` feels like a bit of a code smell resulting from not properly handling the error when it occurred.

Just my opinion :) Happy to chat about more!

Reply:
The following code
https://github.com/kubernetes/kubernetes/blob/8290c85b952a9bcc68c38e8054889a7ff532a0ba/pkg/kubelet/dockershim/convert.go#L68-L73

for the invoke of `toPullableImageID`, the result return nil is acceptable,  the processing here is similar, Therefore, it is not necessary to add a block there.